addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -607,6 +607,13 @@ var FieldChar = InputField.extend(TranslatableFieldMixin, {
         }
         return this._super(value, options);
     },
+
+    isValid: function () {
```

Comment:
Instead of simply marking the field as invalid, it would be handy to have a corresponding mask for the field.
Example: https://imask.js.org/

Reply:
That would be a next step, but these are two different features. The regex validation can be used without a mask.

Reply:
I added that functionality in another branch, I fear that not everyone might like adding a new external library
https://github.com/odoo/odoo/pull/80116

Reply:
you can indeed assert that the field input contains className o_invalid or something. Or at least assert that the view is still in edition, to make it explicit